src/bream/core.py

```diff
@@ -92,67 +92,18 @@ def _is_native_type(spec: TypeSpec) -> bool:
 
 
 @dataclasses.dataclass(frozen=True, slots=True)
-class NoEncoderAvailable:
-    value: object
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class UnencodableDictKey:
-    value: object
-
-
-EncodeError = NoEncoderAvailable | UnencodableDictKey
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class UnsupportedBreamSpec:
-    """The bream spec specified for deserialisation is unsupported."""
-
-    bream_spec: int
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class NoDecoderAvailable:
-    """No decoder is available for the given type_label."""
-
-    type_label: TypeLabel
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class UnsupportedCoderVersion:
+class UnsupportedCoderVersionError(Exception):
```

Comment:
Inheriting from more specialized exceptions would make it easier for users to catch our exceptions.

[`json.decoder.JSONDecodeError`](https://github.com/python/cpython/blob/aeb23273867b27818a3dabd5fca086a1a2e8d229/Lib/json/decoder.py#L20-L43) inherits from `ValueError`. Copy them?

Reply:
Yep, inheriting from `ValueError` crossed my mind too. Let's do that.

Reply:
4417bda